### ISSUE TITLE
Update to latest nearcore stable causes new errors

### ISSUE THREAD

**Issue by emaG3m** [nearprotocol/near-web3-provider#32]
@bowenwang1996 has there been a change in the way transactions are executed? For example:

this call:
```
// Function Call
outcome = await this.account.functionCall(
      this.evm_contract,
       'call_contract',
       { contract_address: utils.remove0x(to), encoded_input: utils.remove0x(data) },
       GAS_AMOUNT.toString(),
       val
);
```

returns this error:
```
InvalidTxError: {}
          at Object.parseRpcError (/Users/emilywilliams/code/summa/near/near-web3-provider/node_modules/near-api-js/lib/utils/rpc_errors.js:38:19)
          at JsonRpcProvider.sendJsonRpc (/Users/emilywilliams/code/summa/near/near-web3-provider/node_modules/near-api-js/lib/providers/json-rpc-provider.js:120:40)
          at processTicksAndRejections (internal/process/task_queues.js:97:5)
          at Account.signAndSendTransaction (/Users/emilywilliams/code/summa/near/near-web3-provider/node_modules/near-api-js/lib/account.js:97:22)
          at NearProvider.routeEthSendTransaction (/Users/emilywilliams/code/summa/near/near-web3-provider/src/index.js:714:27) {
        type: 'InvalidTxError'
      }
```

**Comment by bowenwang1996** [comment]
What is `GAS_AMOUNT`? Please make sure that it is no more than `300000000000000`

**Comment by emaG3m** [comment]
> What is `GAS_AMOUNT`? Please make sure that it is no more than `300000000000000`

THanks @bowenwang1996. Right on the money. After I fixed the gas, I had to fix a few more things. Timestamps now come in as strings rather than numbers. 
Also, when we pass in an invalid hash we used to get a really nice error stating: 
`Failed parsing args: incorrect length for hash`
now it says:
`Failed parsing args: invalid length 2, expected fewer elements in array` 
The old error, from my perspective (using web3) is a lot more helpful than the new one.
